packages/store/addon/-private/utils/promise-record.ts

```diff
@@ -0,0 +1,18 @@
+import InternalModel from '../system/model/internal-model';
+import { promiseObject } from '../system/promise-proxies';
+
+/**
+ * Get the materialized model from the internalModel/promise
+ * that returns an internal model and return it in a promiseObject.
+ *
+ * Useful for returning from find methods
+ *
+ * @private
+ * @param internalModelPromise
+ * @param label
+ */
+export default function promiseRecord(internalModelPromise: Promise<InternalModel>, label: string) {
```

Comment:
`InternalModel` is the wrong type here, we need the `Record` type (which I've created elsewhere, will find)

Reply:
Can you consult @mike-north on this

Reply:
I can//will but it shouldn't block this as I don't add a return type in this PR